### PR TITLE
Place `if` branches on separate lines in more cases

### LINKED ISSUE #75: Better `if` readability
`fmt` currently puts an `if` expression onto one line as long as it will fit in the column limit. However, this can make it tricky to tell where the true branch ends and the false branch begins. See these GitHub threads:

- https://github.com/racket/scribble/pull/441#discussion_r1774250581
- https://github.com/racket/drracket/pull/681#discussion_r1765753280
- https://github.com/racket/drracket/pull/678#discussion_r1757391666

The `if` expressions in those threads are:

- `(if subs (list (restore expr (loop subs))) (list (shift expr)))`
- `(if (< s 0) (loop (+ s (* 2 pi))) s)`
- `(if (boolean? mred-launcher) (if mred-launcher 'mred 'mzscheme) #t)`

I agree with Robby that these are pretty difficult to read. After some discussion in the Racket Discord server, I have a `fmt` proposal: **`if` expressions should only occupy a single line if the true branch and false branch don't have parentheses**. The three examples above would be formatted according to that rule like this:

```
(if subs
    (list (restore expr (loop subs)))
    (list (shift expr)))

(if (< s 0)
    (loop (+ s (* 2 pi)))
    s)

(if (boolean? mred-launcher)
    (if mred-launcher 'mred 'mzscheme)
    #t)
```

However, these `if` expressions would still be allowed to occupy a single line:

- `(if simple? widget gizmo)`
- `(if (foo? x) x #false)`
- `(if (f (g (h a b c))) 'good 'bad)`

This seems to make it easy to tell which expressions are the true and false branches without overly penalizing simple `if` expressions.

### PR DESCRIPTION
**Pull request by 9999years**
This always puts the branches on separate lines, unless all branches are atoms.

Closes #75.

### CODE REVIEW

File: tests/test-cases/large.rkt.out [hunk 1]
```diff
@@ -283,11 +286,14 @@
...
-(provide-class-define-like-keyword [-field field] [-init init] [-init-field init-field])
+(provide-class-define-like-keyword [-field field]
+                                   [-init init]
+                                   [-init-field init-field])
 
 (define-for-syntax not-in-a-class
   (lambda (stx)
-    (if (eq? (syntax-local-context) 'expression)
+    (if (eq? (syntax-local-context)
+             'expression)
```
Comment (jackfirth) [inline tests/test-cases/large.rkt.out:296]: This one isn't quite right. The `(eq? ...)` expression should still be on one line. Not sure why that happened.

File: tests/test-cases/large.rkt.out [hunk 2]
```diff
@@ -334,15 +343,21 @@
 ;; new object struct layer), and this function fills
 ;; in the projections.
 (define (make-field-info cls rpos)
-  (let ([field-ref (make-struct-field-accessor (class-field-ref cls) rpos)]
-        [field-set! (make-struct-field-mutator (class-field-set! cls) rpos)])
+  (let ([field-ref (make-struct-field-accessor (class-field-ref cls)
+                                               rpos)]
+        [field-set! (make-struct-field-mutator (class-field-set! cls)
+                                               rpos)])
```
Comment (jackfirth) [inline tests/test-cases/large.rkt.out:349]: This also has some odd changes.

File: tests/test-cases/large.rkt.out [hunk 3]
```diff
@@ -334,15 +343,21 @@
...
   (let* ([old-ref (unsafe-vector-ref fi 0)]
          [old-set! (unsafe-vector-ref fi 1)])
-    (vector (λ (o) (ppos (old-ref o) neg-party))
-            (λ (o v) (old-set! o (pneg v neg-party)))
+    (vector (λ (o)
+              (ppos (old-ref o)
+                    neg-party))
+            (λ (o v)
+              (old-set! o
+                        (pneg v neg-party)))
```
Comment (jackfirth) [inline tests/test-cases/large.rkt.out:360]: These also should still be one-line. Weird.

**Review by jackfirth — COMMENTED** [review]
I think this is somehow causing a lot more expressions than just `if` to force newlines.

**Comment by sorawee** [comment]
Thank you! Closing this because #80, which is an updated revision of the PR, is merged.

**Pull request closed without merging**